WordPress/gutenberg
# [RNMobile] Search Block: Basic UI

## Description
This PR builds on two previous PRs https://github.com/WordPress/gutenberg/pull/28983 and https://github.com/WordPress/gutenberg/pull/28985 (~those two have to be merged before this one can be taken out of draft~ done!). This PR adds the block UI components and simple styling. Full styling won't be added until a later PR when designs have been finalized. Working features include:
- Displaying and customizing label text. 
- Displaying search input box and either the placeholder text or custom placeholder text.
- Displaying Search button.
- Search button text support and icon only modes now supported in the UI. 
- Hide search label now works

Not included:
- Button position changes do not yet work and will be coming in a future PR. 

https://user-images.githubusercontent.com/5810477/108113864-0b178d00-7066-11eb-9beb-2d51d69138ef.mov

### Gutenberg mobile PR
https://github.com/wordpress-mobile/gutenberg-mobile/pull/3186


#### Known issues
- Button text wrapping acts differently between iOS and Android. This will get addressed in a later PR. 
- Both the search label and search button text use `RichText` and supports formatting but for some odd reason color changes only seem to work in the WordPress app, and not in the demo. Example:

Demo | WordPress
-- | --
![color-text-demo](https://user-images.githubusercontent.com/5810477/108113663-c7248800-7065-11eb-8d4b-af5b907aa8bd.png)|![color-text-wordpress](https://user-images.githubusercontent.com/5810477/108113667-c986e200-7065-11eb-8ef1-e3b9bf608f45.png)

- Button placeholder text is partially hidden

![Screen Shot 2021-02-16 at 2 52 55 PM](https://user-images.githubusercontent.com/5810477/108114329-ab6db180-7066-11eb-845a-c8675a418353.png)


## How has this been tested?
Test the following scenarios and verify the component updates appropriately:
1. Hide and unhide search label by clicking on the option on the toolbar and verify.
2. Toggle icon button only and verify. 
3. Update search label text and verify.
4. update search button text and verify.
5. Update placeholder text and verify.
6. Remove search label text - placeholder should say "Add label..."
7. Remove button text - placeholder should say "Add button text..." (full text not displayed on iOS - known issue)
8. Remove search input placeholder, placeholder should say "Optional placeholder...". 
 

## Types of changes
Part of a new feature to port the Search block to mobile

## Checklist:
- [x] I've tested my changes with keyboard and screen readers.  (Will be handled in a different PR)
